docs/quickstarts/astro.mdx

```diff
@@ -0,0 +1,211 @@
+---
+title: Astro Quickstart
+description: Add authentication and user management to your Astro app with Clerk.
+---
+
+# Astro Quickstart
+
+<TutorialHero
+  framework="astro"
+  beforeYouStart={[
+    {
+      title: "Set up a Clerk application",
+      link: "/docs/quickstarts/setup-clerk",
+      icon: "clerk",
+    },
+    {
+      title: "Create a Astro application",
+      link: "https://docs.astro.build/en/install-and-setup",
+      icon: "astro",
+    },
+  ]}
+  exampleRepo={[
+    {
+      title: "Astro Quickstart Repo",
+      link: "https://github.com/clerk/clerk-astro-quickstart"
+    }
+  ]}
+>
+
+- Install `@clerk/astro`
+- Set up your environment keys to test your app locally
+- Add TypeScript declarations
+- Add the `clerk()` integration to your application
+- Use Clerk middleware to implement route-specific authentication
+- Create a header with Clerk components for users to sign in and out
+
+</TutorialHero>
+
+<Steps>
+### Install `@clerk/astro`
+
+Clerk's [Astro SDK](/docs/references/astro/overview) has prebuilt components and helpers to make user authentication easier.
+
+To get started using Clerk with Astro, add the SDK to your project:
+
+<CodeBlockTabs type="installer" options={["npm", "yarn", "pnpm"]}>
+  ```bash filename="terminal"
+  npm install @clerk/astro
+  ```
+
+  ```bash filename="terminal"
+  yarn add @clerk/astro
+  ```
+
+  ```bash filename="terminal"
+  pnpm add @clerk/astro
+  ```
+</CodeBlockTabs>
+
+### Set environment variables
+
+<SignedIn>
+
+Add the following keys to your `.env.local` file. These keys can always be retrieved from the [API Keys](https://dashboard.clerk.com/last-active?path=api-keys) page of your Clerk Dashboard.
+
+</SignedIn>
+
+<SignedOut>
+
+1. Navigate to the Clerk Dashboard.
+2. In the navigation sidebar, select [API Keys](https://dashboard.clerk.com/last-active?path=api-keys).
+3. In the **Quick Copy** section, copy your Clerk publishable and secret key.
+4. Paste your keys into your `.env.local` file.
+
+The final result should resemble the following:
+
+</SignedOut>
+
+  <InjectKeys>
+
+  ```sh filename=".env"
+  PUBLIC_CLERK_PUBLISHABLE_KEY={{pub_key}}
+  CLERK_SECRET_KEY={{secret}}
+  ```
+
+  </InjectKeys>
+
+
+### Add TypeScript declarations
+
+Inside your `src/` directory, update the `.env.d.ts` file to ensure that the locals added by the middleware are properly typed.
```

Comment:
```suggestion
Inside your `src/` directory, update the `.env.d.ts` file to define types for the local variables added by the middleware.
```

Reply:
Seems like the Astro docs refer to "local variables" as any variable within an astro file, which is definitely not what "locals" are. 
Also types are not just for locals in the middleware, because they are accessible from any astro page/endpoint.

Reply:
Great points. I edited the sentence to avoid confusion with local variables and to start with an action verb.
```suggestion
Update the `.env.d.ts` file in your `src/` directory to add type definitions for the `locals` object added by the middleware. 
```

Reply:
I like the suggestion of starting with an action verb, committing it!